### PR TITLE
Remove GCurrentTimeGraph global.

### PR DESCRIPTION
**Pull request by florian-kuebler**
Effectively the global GCurrentTimeGraph is always
app_::capture_window_::time_graph_ (with the fuzzer being the only
exception). This change removes the global and use the value directly
instead.
It also fixes crashes in the fuzzer, which also works with this change
due to the check if capture_window_ exists.

Follow-up: There is an existing bug, that shows the timerange in
the capture window after clearing a capture. We on clearing a capture,
we should not "clear" the time graph, but actually remove it.

Test: Take, clear, load captures.
Bug: http://b/177651082

### CODE REVIEW

**Review by dimitry- — APPROVED** [review]

File: src/OrbitGl/App.cpp [hunk 1]
```diff
@@ -909,17 +906,15 @@ void OrbitApp::AbortCapture() {
 
 void OrbitApp::ClearCapture() {
   ORBIT_SCOPE_FUNCTION;
-  capture_window_->GetTimeGraph()->SetCaptureData(nullptr);
```
Comment (florian-kuebler) [inline src/OrbitGl/App.cpp:912]: Note, that the "set capture data" now happens inside the "clear".

File: src/OrbitGl/App.cpp [hunk 2]
```diff
@@ -1166,42 +1161,41 @@ void OrbitApp::LoadSymbols(const std::filesystem::path& symbols_path, ModuleData
...
+        }
+        LOG("Added frame tracks in module \"%s\"", module_data->file_path());
+      }
 
-          UpdateAfterSymbolLoading();
-          FireRefreshCallbacks();
-        });
+      UpdateAfterSymbolLoading();
+      FireRefreshCallbacks();
+    });
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/OrbitGl/App.cpp:1198]: That is feeling strange. I do clang-format all the files that I touch, so it might be an issue here.

[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/OrbitGl/App.cpp:1198]: It looks correct and clang-format check accepts it. So I'd just keep it.

File: src/OrbitGl/App.cpp [hunk 3]
```diff
@@ -1166,42 +1161,41 @@ void OrbitApp::LoadSymbols(const std::filesystem::path& symbols_path, ModuleData
...
+                                         std::move(frame_track_function_hashes)] {
+      modules_currently_loading_.erase(module_data->file_path());
+
+      const ProcessData* selected_process = GetTargetProcess();
+      if (selected_process != nullptr &&
+          selected_process->IsModuleLoaded(module_data->file_path())) {
+        functions_data_view_->AddFunctions(module_data->GetFunctions());
+        LOG("Added loaded function symbols for module \"%s\" to the functions tab",
+            module_data->file_path());
+      }
```
Comment (beckerhe) [inline src/OrbitGl/App.cpp:1176]: I can't see what has changed here. Am I blind?

File: src/OrbitGl/App.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (beckerhe) [reply src/OrbitGl/App.cpp:1198]: Could you put that into a separate commit? It will make rebasing for others a lot easier.

File: src/OrbitGl/App.h [hunk 4]
```diff
@@ -145,6 +145,14 @@ class OrbitApp final : public DataViewFactory, public CaptureListener {
...
 
   void SetCaptureWindow(CaptureWindow* capture);
+  [[nodiscard]] const TimeGraph* GetTimeGraph() const {
+    CHECK(capture_window_ != nullptr);
+    return capture_window_->GetTimeGraph();
+  }
+  [[nodiscard]] TimeGraph* GetMutableTimeGraph() {
+    CHECK(capture_window_ != nullptr);
+    return capture_window_->GetTimeGraph();
+  }
```
Comment (beckerhe) [inline src/OrbitGl/App.h:155]: There is an `if (capture_window != nullptr)` somewhere in the code. That tells me that we can't assume that `capture_window_` is always existing. Have you considered returning a `std::optional<TimeGraph*>` and not abort if `capture_window_` does not exist?

File: src/OrbitGl/LiveFunctionsController.cpp [hunk 5]
```diff
@@ -35,9 +35,8 @@ std::pair<uint64_t, uint64_t> ComputeMinMaxTime(
 uint64_t AbsDiff(uint64_t a, uint64_t b) {
   if (a > b) {
     return a - b;
-  } else {
-    return b - a;
   }
+  return b - a;
```
Comment (beckerhe) [inline src/OrbitGl/LiveFunctionsController.cpp:39]: Nit: This is unrelated and should be a separate commit.

File: src/OrbitGl/LiveFunctionsController.cpp [hunk 6]
```diff
@@ -49,22 +48,22 @@ const TextBox* ClosestTo(uint64_t point, const TextBox* box_a, const TextBox* bo
   return box_b;
 }
 
-const TextBox* SnapToClosestStart(uint64_t function_id) {
-  double min_us = GCurrentTimeGraph->GetMinTimeUs();
-  double max_us = GCurrentTimeGraph->GetMaxTimeUs();
+const TextBox* SnapToClosestStart(const OrbitApp* app, uint64_t function_id) {
```
Comment (beckerhe) [inline src/OrbitGl/LiveFunctionsController.cpp:51]: When is `OrbitApp` actually needed? As far as I can see a `const TimeGraph*` would be enough.

**Review by beckerhe — COMMENTED** [review]
Sorry for being late. I added some comments. I hope it's not too late. :see_no_evil:

File: src/OrbitGl/App.cpp (hunk 3 above)
Comment (beckerhe) [reply src/OrbitGl/App.cpp:1176]: Nevermind. See other comment.

File: src/OrbitGl/App.h (hunk 4 above)
Comment (florian-kuebler) [reply src/OrbitGl/App.h:155]: Well, yeah I though of this. But in the code there is exactly one place where we do the capture_window != nullptr check, on all other places we currently assume that GCurrentTimeGraph is not null (actually without stating it explicitly). I am not sure if we want to add handling for " no capture window present" for all the callers, or if we would end up in adding CHECKs on every caller, in which case we could also just keep it as it is now. It is an improvement to the current state, as we at least do have the checks here.

But I am happy to discuss, and don't have strong feelings about this.

Nit question: Assume we do handle this case here, why not just returning nullptr?

Comment (beckerhe) [reply src/OrbitGl/App.h:155]: I see your point. Before we ignored it, but it wasn't so obvious. Now, we keep the ignoring part but at least improve in other parts. So I guess it's fine for me to leave it.

Regarding nullptr: I prefer not to return a nullptr, but rather make a syntactical change. We had a lot of crashes over the last few releases which were related to similar getters. `GetProcess`, `GetProcessManager`, `GetModuleByPath`. You never know whether you can assume the returned pointer to be valid or not. And someone might change the behaviour and it will still compile. That's why I advocate for a raw pointer to be never null. If you need an empty value, wrap it into a std::optional.

I want to make a small presentation about that because I feel that is related to a lot of crashes in the past. But of course that shouldn't stop you now. :-)

File: src/OrbitGl/LiveFunctionsController.cpp (hunk 6 above)
Comment (florian-kuebler) [reply src/OrbitGl/LiveFunctionsController.cpp]: Done.

File: src/OrbitGl/LiveFunctionsController.cpp (hunk 5 above)
Comment (florian-kuebler) [reply src/OrbitGl/LiveFunctionsController.cpp:39]: Done.

File: src/OrbitGl/App.h (hunk 4 above)
Comment (florian-kuebler) [reply src/OrbitGl/App.h:155]: Okay, then lets leave it for now.

File: src/OrbitGl/App.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/OrbitGl/App.cpp]: Should be obsolete now.

**Review by florian-kuebler — COMMENTED** [review]
Addressed the comments. Thanks for those.

**Review by beckerhe — APPROVED** [review]
Thank you Florian!

File: src/OrbitGl/LiveFunctionsController.cpp [hunk 7]
```diff
@@ -217,12 +217,18 @@ uint64_t LiveFunctionsController::GetStartTime(uint64_t index) {
   return GetCaptureMin();
 }
 
-uint64_t LiveFunctionsController::GetCaptureMin() { return GCurrentTimeGraph->GetCaptureMin(); }
-uint64_t LiveFunctionsController::GetCaptureMax() { return GCurrentTimeGraph->GetCaptureMax(); }
+uint64_t LiveFunctionsController::GetCaptureMin() {
```
Comment (karupayun) [inline src/OrbitGl/LiveFunctionsController.cpp:220]: const and [[nodiscard]]?

File: src/OrbitGl/LiveFunctionsController.cpp [hunk 8]
```diff
@@ -217,12 +217,18 @@ uint64_t LiveFunctionsController::GetStartTime(uint64_t index) {
...
+  CHECK(app_ != nullptr);
+  return app_->GetTimeGraph()->GetCaptureMin();
+}
+uint64_t LiveFunctionsController::GetCaptureMax() {
```
Comment (karupayun) [inline src/OrbitGl/LiveFunctionsController.cpp:224]: const and [[nodiscard]]?

File: src/OrbitGl/LiveFunctionsController.cpp (hunk 7 above)
Comment (florian-kuebler) [reply src/OrbitGl/LiveFunctionsController.cpp:220]: Good catch. I can do a follow-up PR on Monday. There were quite a bit of clang-tidy warnings in this file.
